pyspark/dl/optim/optimizer.py

```diff
@@ -188,6 +188,14 @@ def setvalidation(self, batch_size, val_rdd, trigger, val_method=["Top1Accuracy"
         callBigDlFunc(self.bigdl_type, "setValidation", self.value, batch_size,
                       trigger, val_rdd, val_method)
 
+    def setmodel(self, model):
```

Comment:
I didn't add a separate method perpare_input as I think the concern here is when create optimizer, it silently load/caching data which makes user confuse why create optimizer takes so long. I think it should be fine if we explicit told user we are loading data. Besides I think prepare_input is used to tell user we are loading data. I am not very likely to add the method and ask user call it. But if most of you think it's necessary to add the method, I will do it.